### PR TITLE
repo: expand PR template with checklist

### PR DESCRIPTION
**Pull request by raduweiss**
## Reason for This PR

To ensure that the Firecracker docs and changelog are updated with each PR
where that's needed, the PR template Is extended with an appropriate checklist.

## Description of Changes

The PR template now includes a checklist that encourages both author and
reviewer to verify that:
- The PR's commits are signed.
- The PR is well reasoned.
- The PR has a clear description.
- The PR includes documentation updates (if needed).
- The PR includes swagger API definition updates (if needed).
- The PR includes changelog updates (if needed).

 ## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

 ## PR Checklist

`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (git commit -s).
- [x] The reason for this PR is clearly provided.
- [x] The description of changes is clear and encompassing.
- [x] No docs need to be updated as part of this PR.
- [x] No code has been touched
- [x] No API changes are included in this PR.
- [x] The changes in this PR have no user impact.

### CODE REVIEW

**Comment by acatangiu** [comment]
I suggest we also add a checkbox for "Make sure all commits are signed (`git commit -s`)".

**Comment by kzys** [comment]
@acatangiu FYI, firecracker-containerd is doing the check as a part of CI on BuildKite.

https://github.com/firecracker-microvm/firecracker-containerd/blob/2905f8dbab16c6847c99dd6384a90b462bffe206/.buildkite/logcheck.sh#L31

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Reason for This PR
+
+`[Author TODO: add issue # of explain reasoning.]`
```
Comment (dianpopa) [inline .github/pull_request_template.md]:
```suggestion
`[Author TODO: add issue # or explain reasoning.]`
```

**Comment by dianpopa** [comment]
> @acatangiu FYI, firecracker-containerd is doing the check as a part of CI on BuildKite.
> 
> https://github.com/firecracker-microvm/firecracker-containerd/blob/2905f8dbab16c6847c99dd6384a90b462bffe206/.buildkite/logcheck.sh#L31

We now have #1231 to remember us of this.

**Comment by raduweiss** [comment]
> @acatangiu FYI, firecracker-containerd is doing the check as a part of CI on BuildKite.
> 
> https://github.com/firecracker-microvm/firecracker-containerd/blob/2905f8dbab16c6847c99dd6384a90b462bffe206/.buildkite/logcheck.sh#L31

That's clearly the right way to do it, but until we're there, the checklist will have to do :)

**Comment by raduweiss** [comment]
> Other two possible checklist items:
> [ ] Check that changes do not modify API request syntax otherwise firecracker/swagger.yaml (**insert link to file here**) is updated.
> [ ] Code needs relevant metrics added (especially if device code is modified).
> (Not very sure about the last one, we may sound too demanding...)

I picked up the API suggestion. For metrics, we need to make a very simple statement that allows contributors and reviewers to decide on the checkbox item, but I don't think things are so clear-cut now.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
+`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`
+
+- [ ] All commits in this PR are signed (`git commit -s`).
+- [ ] The PR is either linked to an issue, or a reason is provided. 
+- [ ] The description of changes is clear and encompassing.
+- [ ] Either no docs need to be updated as part of this PR, or, the required
+      changes are included in this PR. Docs in scope are all `*.md` files
+      located either in the repository root, or in the `docs/` directory.
```
Comment (andreeaflorescu) [inline .github/pull_request_template.md]: We also have documentation in the root of each crate for some crates. One example is micro_http where recently we had a PR that didn't update that documentation. Should we also mention this type documentation (in-crate documentation)?

**Pull request merged**